code/modules/clothing/spacesuits/_spacesuits.dm

```diff
@@ -199,11 +200,15 @@
 	// If we're turning thermal protection on, check for valid cell and for enough
 	// charge that cell. If it's too low, we shouldn't bother with setting the
 	// thermal protection value and should just return out early.
+	var/mob/living/carbon/human/user = src.loc
 	if(!thermal_on && !(cell && cell.charge >= THERMAL_REGULATOR_COST))
+		to_chat(user, "<span class='warning'>The thermal regulator on \the [src] has no charge.</span>")
 		return
 
 	thermal_on = !thermal_on
 	min_cold_protection_temperature = thermal_on ? SPACE_SUIT_MIN_TEMP_PROTECT : SPACE_SUIT_MIN_TEMP_PROTECT_OFF
+	if(user || ishuman(user) || (user.wear_suit == src))
```

Comment:
if user exists, or if user is a human typepath, or if the user is wearing the suit.

ARCANEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE